ansible/roles/nginx/tasks/gcs.yml

```diff
@@ -0,0 +1,59 @@
+---
+
+- name:  install necessary packages for Gcloud
+  apt:
+    name:
+      - apt-transport-https
+      - ca-certificates
+      - gnupg
+    state: present
+
+- name: Add GPG key for Gcloud repository
+  shell: "curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -"
+
+- name: Add Gcloud repository (Debian/Ubuntu)
+  apt_repository:
+    repo: "{{ gcloud_apt_deb_ubuntu_repository }}"
+    state: present
+    update_cache: true
+  when: ansible_distribution == "Debian"
+    or ansible_distribution == "Ubuntu"
+
+- name: install Gcloud package
+  apt:
+    name: google-cloud-cli
+    state: present
+    update_cache: true 
+      
+- name: create GCS folders
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: 0755
+    recurse: true
+  with_items:
+    - "{{ nginx_gcs_workdir }}"
+    - "{{ nginx_keys_workdir }}"
+
+- name: Copy GCP service account file
+  copy:
+    src: "{{ gcp_service_account_host_file }}"
+    dest: "{{ nginx_keys_workdir }}/key.json"
+    owner: '1000'
+    group: '1000'
+    mode: 0644
+
+- name: Copy GCS rsync script
+  template:
+    src: gcs_rsync.sh.j2
+    dest: "{{ gcs_rsync_script }}"
+    mode: 0500
+    owner: "{{ ansible_user_id }}"
+
+- name: Activate service account
+  shell: gcloud auth activate-service-account --key-file "{{ nginx_keys_workdir }}/key.json"
+   
+- name: Launch first rsync
```

Comment:
Instead of running a script here, why don't we run the command that it's on the script directly?

Reply:
Also, maybe it makes sense to move the `rsync` outside this role and put it into the main file, so the role, only installs `gcloud`.